source/Halibut.TestUtils.CompatBinary.Base/ProxyServiceForwardingRequestToClient.cs

```diff
@@ -0,0 +1,124 @@
+    /// has the old client which is making requests to the new service in the test CLR.
+    /// The Proxy 
+    /// </summary>
+    public class ProxyServiceForwardingRequestToClient
+    {
+        public static int Run(string[] args)
+        {
+            var tentacleCertPath = Environment.GetEnvironmentVariable("tentaclecertpath");
+            Console.WriteLine($"Using tentacle cert path: {tentacleCertPath}");
+            var serviceCert = new X509Certificate2(tentacleCertPath);
+            
+            Console.WriteLine("Tentacle/service cert details " + serviceCert);
+
+            
+            var octopusCertPath = Environment.GetEnvironmentVariable("octopuscertpath");
+            Console.WriteLine($"Using octopus cert path: {octopusCertPath}");
+            var clientCert = new X509Certificate2(octopusCertPath);
+            Console.WriteLine("Octopus/Client cert details " + clientCert);
+            
+
+            ServiceConnectionType serviceConnectionType = BackwardsCompatProgramBase.ServiceConnectionTypeFromString(Environment.GetEnvironmentVariable("ServiceConnectionType"));
+            string addressToPoll = null;
+            if (serviceConnectionType == ServiceConnectionType.Polling)
+            {
+                addressToPoll = Environment.GetEnvironmentVariable("octopusservercommsport");
+                Console.WriteLine($"Will poll: {addressToPoll}");
+            }
+
+            string serivceAddressToConnectTo = null; 
+            if (serviceConnectionType == ServiceConnectionType.Listening)
+            {
+                serivceAddressToConnectTo = Environment.GetEnvironmentVariable("realServiceListenAddress");
+                Console.WriteLine($"Will forward request to: {serivceAddressToConnectTo}");
+            }
+            
+            // Create a client which will send RPC calls to the service in the Test CLR.
+            Console.WriteLine("This should be bob:");
+            Console.WriteLine(clientCert.ToString());
+            using (var client = new HalibutRuntimeBuilder()
+                       .WithServerCertificate(clientCert)
+                       .WithLogFactory(new TestContextLogFactory("ProxyClient"))
+                       .Build())
+            {
+                Console.WriteLine("Next line should be: 36F35047CE8B000CF4C671819A2DD1AFCDE3403D");
+                Console.WriteLine("Client will trust: " + serviceCert.Thumbprint);
+                client.Trust(serviceCert.Thumbprint);
+
+                
+                Uri realServiceUri = null;
+                switch (serviceConnectionType)
+                {
+                    case ServiceConnectionType.Polling:
+                        var clientPollingListeningPort = client.Listen();
+                        Console.WriteLine("Polling listener is listening on port: " + clientPollingListeningPort);
+                        
+                        // This needs to be done within the CLR.
```

Comment:
??

Reply:
I think they are notes for me when I was converting it. I will remove.